experimental/dds/tree2/src/shared-tree/sharedTree.ts

```diff
@@ -141,6 +145,37 @@ export class SharedTree
 			events: this._events,
 			removedTrees,
 		});
+		this.viewForSummarization = this.view;
+
+		let activeTransactionCount = 0;
+
+		const transactionStart = () => {
+			if (activeTransactionCount === 0) {
+				this.viewForSummarization = this.view.fork();
+			}
+			activeTransactionCount++;
+		};
+		const transactionFinish = () => {
+			activeTransactionCount--;
+			if (activeTransactionCount === 0) {
+				this.viewForSummarization.dispose();
+				this.viewForSummarization = this.view;
+			}
+		};
+		const deregisterStart = this.view.transaction.events.on("start", transactionStart);
+		const deregisterAbort = this.view.transaction.events.on("abort", transactionFinish);
+		const deregisterCommit = this.view.transaction.events.on("commit", transactionFinish);
+		this.deregisterViewClone = () => {
+			deregisterStart();
+			deregisterAbort();
+			deregisterCommit();
+			this.deregisterViewClone = undefined;
```

Comment:
technically shouldn't be necessary but `didAttach` gets fired twice in some scenarios. I filed AB#5955 for that.